src/grt/src/fastroute/src/FastRoute.cpp

```diff
@@ -682,6 +684,98 @@ NetRouteMap FastRouteCore::getRoutes()
   return routes;
 }
 
+NetRouteMap FastRouteCore::getPartialRoutes()
+{
+  NetRouteMap routes;
+  bool is_vertical = ((2 % 2) - layer_orientation_) != 0;
+
+  for (int netID = 0; netID < netCount(); netID++) {
+    auto fr_net = nets_[netID];
+    odb::dbNet* db_net = fr_net->getDbNet();
+    GRoute& route = routes[db_net];
+    std::unordered_set<GSegment, GSegmentHash> net_segs;
+
+    const auto& treeedges = sttrees_[netID].edges;
+    const int num_edges = sttrees_[netID].num_edges();
+
+    for (int edgeID = 0; edgeID < num_edges; edgeID++) {
+      const TreeEdge* treeedge = &(treeedges[edgeID]);
+      if (treeedge->len > 0) {
+        int routeLen = treeedge->route.routelen;
+        const std::vector<short>& gridsX = treeedge->route.gridsX;
+        const std::vector<short>& gridsY = treeedge->route.gridsY;
+        const std::vector<short>& gridsL = treeedge->route.gridsL;
+        int lastX = tile_size_ * (gridsX[0] + 0.5) + x_corner_;
+        int lastY = tile_size_ * (gridsY[0] + 0.5) + y_corner_;
+        int lastL_h;
+        int lastL_v;
+        if (gridsL.empty()) {
+          if (is_vertical) {
+            lastL_h = 1;
+            lastL_v = 2;
```

Comment:
What are 1 & 2?  Are these directions?

Reply:
These are the layers definition for vertical edges and horizontal edges

Reply:
lastL_h and lastL_v were renamed to layer_h and layer_v respectively.